src/Transport/Creation/AzureServiceBusQueueCreator.cs

```diff
@@ -31,7 +31,7 @@ public AzureServiceBusQueueCreator(TopologyQueueSettings queueSettings, ReadOnly
                 DefaultMessageTimeToLive = queueSettings.DefaultMessageTimeToLive,
                 EnableDeadLetteringOnMessageExpiration = queueSettings.EnableDeadLetteringOnMessageExpiration,
                 DuplicateDetectionHistoryTimeWindow = queueSettings.DuplicateDetectionHistoryTimeWindow,
-                MaxDeliveryCount = IsSystemQueue(queuePath) ? 10 : numberOfImmediateRetries,
+                MaxDeliveryCount = DefaultMaxDeliveryCountForNoImmediateRetries,
```

Comment:
default value will disable dlq basically, is that what we want?

Reply:
Does it actually disable it, as in we could not use it to send non deserializable messages? or does it just "effectively disabled" meaning retries will effectively go on forever?

If it is the first, we should set it to some large number that doesn't disable the DLQ. If it is the second then that is the desired result.

Reply:
We try to calculate the next to max # of receive value for `MaxDeliveryCount` on the entities to ensure ASB's dead-lettering doesn't kick in and allow NSB's recoverability to work properly. I.e. need to ensure DLQ feature is "deactivated". Except we can't. Setting it to `int.MaxValue` does exactly that and indicates that the value is not a user arbitrary one, but set by NSB. No sane person will use this large value :slightly_smiling_face:

Thinking of it, we solely make use of DLQ when _ deliberately_ send there a message that cannot be de-serialized. We never want messages to be sent to DLQ as we have the site queue concept and it's mandatory.

DLQ is a native error queue, in some way. NServiceBus error queue mechanism should take the precedence.

Reply:
I understand that, but the word "disable" @yvesgoeleven used implies an inability to use DLQ, which as you said we use for poison non-deserializable messages.

If it isn't disabled in the sense that it can still be used but the native retry mechanism is irrelevant, then that is the desired goal and the `int.MaxValue` is a good value. If there is undocumented functionality that when set to `int.MaxValue` disables the queue so it cannot be used for poison messages, then we should pick another arbitrarily large and insane value, but that does not trigger this undocumented functionality.

Reply:
@boblangley it's not disabled in that sense afaik, will still be able to manually deadletter

Reply:
Correct @yvesgoeleven. It's "muted". Still can dead-letter explicitly. For example: messages that fail de-serialization will be dead-lettered by our code.